src/main/scala/interfaces/Verification.scala

```diff
@@ -21,25 +21,22 @@ import viper.silicon.state.terms.Term
 
 /* TODO: Make VerificationResult immutable */
 sealed abstract class VerificationResult {
-  var previous: Option[NonFatalResult] = None
+  var previous: Set[VerificationResult] = Set()
```

Comment:
Keep in mind that sets typically have non-deterministic iteration order, which may be undesirable (e.g. maybe we want earlier results to also occur earlier in the collection?) or confusing (e.g. when outputting the results without sorting them first).

If necessary, alternatives would be set instance that preserves insertion order or are sorted/ordered, or using another data structure, e.g. a vector.

Reply:
As discussed, this will be changed if/when it will be needed.